### PR TITLE
added pagination, sorting and swagger

### PR DESCRIPTION
**Pull request by esidarapp**

### CODE REVIEW

File: src/main/java/com/example/spring/controller/BookController.java [hunk 1]
```diff
@@ -18,38 +21,45 @@
...
 @RestController
 @RequestMapping("/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> findAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> findAll(Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/controller/BookController.java]:
```suggestion
    public List<BookDto> findAll(@ParameterObject @PageableDefault Pageable pageable) {

```

File: src/main/java/com/example/spring/controller/BookController.java [hunk 2]
```diff
@@ -18,38 +21,45 @@
...
 
     @PostMapping
+    @Operation(summary = "Create a new book", description = "Create a new book")
     public BookDto save(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update book by id", description = "Update book by id")
     public BookDto updateById(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/controller/BookController.java]:
```suggestion
    public BookDto updateById(@PathVariable Long id, @RequestBody @Valid CreateBookRequestDto requestDto) {

```

File: src/main/java/com/example/spring/controller/BookController.java [hunk 3]
```diff
@@ -18,38 +21,45 @@
...
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
+    @Operation(summary = "Delete book by id", description = "Delete book by id")
     public void deleteById(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
+    @Operation(summary = "Search books by params", description = "Search books by params")
     @GetMapping("/search")
     public List<BookDto> search(BookSearchParameters searchParameters) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/controller/BookController.java]:
```suggestion
    public List<BookDto> search(BookSearchParameters searchParameters, @ParameterObject @PageableDefault Pageable pageable) {

```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**